api/commands/format.ts

```diff
@@ -0,0 +1,182 @@
+import { constructSyntaxTree, ISyntaxTreeNode } from "@dataform/sqlx/lexer";
+import * as crypto from "crypto";
+import * as fs from "fs";
+import * as jsBeautify from "js-beautify";
+import * as sqlFormatter from "sql-formatter";
+import { promisify } from "util";
+
+const JS_BEAUTIFY_OPTIONS: JsBeautifyOptions = {
+  indent_size: 2,
+  preserve_newlines: false
+};
+
+export async function formatFile(
+  filename: string,
+  options?: {
+    overwriteFile?: boolean;
+  }
+) {
+  const fileText = await promisify(fs.readFile)(filename, "utf8");
+  const fileExtension = filename.split(".").slice(-1)[0];
+  const formattedText = (() => {
+    try {
+      switch (fileExtension) {
+        case "sqlx":
+          return formatSqlx(constructSyntaxTree(fileText));
+        case "js":
+          return formatJavaScript(fileText);
+        default:
+          return fileText;
+      }
+    } catch (e) {
+      throw new Error(`Unable to format "${filename}": ${e.message}`);
+    }
+  })();
+  if (options && options.overwriteFile) {
+    await promisify(fs.writeFile)(filename, formattedText);
+  }
+  return formattedText;
+}
+
+function formatSqlx(node: ISyntaxTreeNode) {
+  return postProcessFormattedSqlx(
+    getIndividualSqlxStatements(node.contents)
+      .map(individualStatement => {
+        const placeholders: {
+          [placeholderId: string]: ISyntaxTreeNode;
+        } = {};
+        const unformattedPlaceholderSql = individualStatement
+          .map(child => {
+            if (typeof child !== "string") {
+              const placeholderId = crypto.randomBytes(16).toString("hex");
```

Comment:
Can we seed this easily so it's consistent?

Reply:
I couldn't find a way to seed random numbers in node short of depending on another package :(